### PR TITLE
Routing | AddTodo | DisplayTask

### PR DESCRIPTION
**Pull request by shubhmore1212**

### CODE REVIEW

File: src/CustomHooks/useFetch.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from "react";
+
+export const useFetch = (url: string) => {
+  const [data,setData]=useState<any>(null);
+  const [loading,setLoading]=useState<boolean>(true);
+  const [error,setError]=useState("");
```
Comment (vaibhavchalse99) [inline src/CustomHooks/useFetch.ts]: type ?

File: src/CustomHooks/useFetch.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+   useEffect(()=>{
+    fetch(url)
+      .then((res) => {
+        if (!res.ok) throw new Error("Could not fetch data");
+        return res.json();
+      })
+      .then((res) =>
+        setTimeout(() => {
+          //Spliced only for todo since the list is big
+            setData(res);
```
Comment (vaibhavchalse99) [inline src/CustomHooks/useFetch.ts]: check indentation

File: src/Modules/AddTodo/components/index.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+interface IProps {
+  title: string;
+  setTitle: (title: string) => void;
+  body: string;
+  setBody: (body: string) => void;
+  handleSubmit: (e: any) => void;
+}
+
+const AddTodoComponent: React.FC<IProps> = ({
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: destructure in the component

File: src/Modules/AddTodo/components/index.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  title,
+  setTitle,
+  body,
+  setBody,
+  handleSubmit,
+}) => {
+  return (
+    <div className="add-form">
+      <form onSubmit={(e) => handleSubmit(e)}>
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: (e)=> handleSubit(e) can be replaced with handleSubmit

File: src/Modules/AddTodo/components/index.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        <h2>Add Todo</h2>
+        <div>
+          <label htmlFor="title">Task:</label>
+          <input
+            type="text"
+            name="title"
+            value={title}
+            onChange={(e) => setTitle(e.target.value)}
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: write a separate handler and set the state inside that handler.

File: src/Modules/AddTodo/components/index.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+          />
+          <br />
+        </div>
+        <div>
+          <label htmlFor="body">Details:</label>
+          <textarea
+            name="body"
+            value={body}
+            onChange={(e) => setBody(e.target.value)}
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: +1

File: src/Modules/AddTodo/components/index.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+          />
+          <br />
+        </div>
+        <div>
+          <label htmlFor="completed">Status:</label>
+          <input type="text" defaultValue={"false"} disabled />
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: default value should be "Pending"

why di we need brackets?

File: src/Modules/AddTodo/components/index.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+          <br />
+        </div>
+        <input type="submit" value="Add Todo" />
+      </form>
+      <NavLink className={"link-home"} to="/">{`<< Go to ToDo-Page`}</NavLink>
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: {"link-home"}  should be "link-home"

Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: use constant variables that we have created for routes for "/" as HOME

File: src/Modules/AddTodo/index.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
+import React, { useState } from "react";
+
+import AddTodoComponent from "./components"
+
+import { TODO_URL } from "../DisplayTodo/constant";
+
+const AddTodoContainer=()=>{
+    const [title, setTitle] = useState<string>("");
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/index.tsx]: indentation?

File: src/Modules/AddTodo/index.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  const [body, setBody] = useState<string>("");
+
+  const handleSubmit = async (e: any) => {
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/index.tsx]: check the type we can assign for this event.
It should have a proper type

File: src/Modules/AddTodo/index.tsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    e.preventDefault();
+    let tempObj = {};
+    if (
+      title?.trim() !== "" &&
+      title !== undefined &&
+      body?.trim() !== "" &&
+      body !== undefined
+    ) {
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/index.tsx]: this condition can be used as 
if( ! title?.trim() && !body?.trim())

File: src/Modules/AddTodo/index.tsx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+      let resData = await fetch(TODO_URL);
+      let res = await resData.json();
+
+      tempObj = {
+        ...tempObj,
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/index.tsx]: remove ...tempObj

File: src/Modules/AddTodo/index.tsx [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+        id: res[res.length-1]["id"]+1,
+        title: title,
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/index.tsx]: if key and value is same use it like below
title: title, -----> title,

File: src/Modules/AddTodo/index.tsx [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+      title !== undefined &&
+      body?.trim() !== "" &&
+      body !== undefined
+    ) {
+      let resData = await fetch(TODO_URL);
+      let res = await resData.json();
+
+      tempObj = {
+        ...tempObj,
+        id: res[res.length-1]["id"]+1,
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/index.tsx]: make (res[res.length-1]?.id || 0 )+1

File: src/Modules/AddTodo/index.tsx [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+    setBody("");
+  };
+
+  return (
+    <AddTodoComponent
+    title={title}
+    setTitle={setTitle}
+    body={body}
+    setBody={setBody}
+     handleSubmit={handleSubmit}
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/index.tsx]: indentation?

File: src/Modules/AddTodo/index.tsx [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(tempObj),
+      })
+        .then((res) => res.json())
+        .catch((err) => console.log(err.message));
+    }
+
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/index.tsx:67]: remove black line

File: src/Modules/DisplayTask/components/index.tsx [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+
+interface IProps{
+    data:ToDoData,
+    loading:boolean,
+    error:string,
+    markTodoCompleted:(completed:string)=>void,
+    deleteTodo:(id:number)=>void
+}
+
+const DisplayTaskComponent :React.FC<IProps> = ({data,loading,error,markTodoCompleted,deleteTodo}) => {
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: destructure in the component

File: src/Modules/DisplayTask/components/index.tsx [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+  return (
+    <>
+      {loading ? (
+        <div className="loading-spinner">
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: create a separate component for this spinner

File: src/Modules/DisplayTask/components/index.tsx [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+          <InfinitySpin color="rgb(246, 162, 67)" width="200" />
+        </div>
+      ) : (
+        <>
+          {error === "" ? (
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: use !error

File: src/Modules/DisplayTask/components/index.tsx [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+            <div className="display-table">
+              <h1>To Do</h1>
+              <div className="cross-button">
+                <NavLink
+                  className="cross-button-link"
+                  title="Go to ToDo Page"
+                  to="/"
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: use constant from routes

File: src/Modules/DisplayTask/components/index.tsx [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+                  <tr>
+                    <th>Body :</th>
+                    <td>{data.body}</td>
+                  </tr>
+                  <tr>
+                    <th>Status :</th>
+                    <td>
+                      <select
+                        defaultValue={
+                          (data.completed as boolean) ? "Completed" : "Pending"
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: create a constants for Completed and pending status

File: src/Modules/DisplayTask/components/index.tsx [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+                        }
+                        onChange={(e) => markTodoCompleted(e.target.value)}
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: create a separate handler and set the state there inside that handler

File: src/Modules/DisplayTask/index.tsx [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+import { useFetch } from "../../CustomHooks/useFetch";
+
+import { TODO_URL } from "../DisplayTodo/constant";
+
+const DisplayTaskContainer = () => {
+  const { id } = useParams();
+  const { data, loading, error } = useFetch(TODO_URL + `/${id}`);
+  const navigate = useNavigate();
+
+  const deleteTodo = (todoId: any) => {
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/index.tsx]: give proper id type

File: src/Modules/DisplayTask/index.tsx [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+    fetch(`${TODO_URL}/${todoId}`, {
+      method: "Delete",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    })
+      .then((res) => res.json())
+      .catch((err) => console.log(err));
+    navigate("/");
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/index.tsx]: use Route constant

File: src/Modules/DisplayTask/index.tsx [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(tempObj),
+    })
+      .then((res) => res.json())
+      .catch((err) => console.log(err));
+  };
+
+  return <DisplayTaskComponent 
+  data={data}
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/index.tsx]: indentation?

File: src/Modules/DisplayTodo/components/AddTodoButton.tsx [hunk 26]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+const AddTodoButton = () => {
+  return (
+    <div className="d-flex">
+      <Link to="/add-todo">
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/components/AddTodoButton.tsx]: use constant variable

File: src/Modules/DisplayTodo/components/TodoList.tsx [hunk 27]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+
+import TodoListItem from "./TodoListItem";
+
+import { ToDoData } from "../../../data/types/types";
+
+const TodoList = (props: {
+  todos: ToDoData[];
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/components/TodoList.tsx:8]: destructure inside the component

File: src/Modules/DisplayTodo/components/TodoListItem.tsx [hunk 28]
```diff
@@ -0,0 +1,34 @@
...
+
+interface IProps {
+  id: number;
+  title: string;
+  completed: boolean;
+  markTodoCompleted: (id: number, completed: boolean) => void;
+}
+
+const TodoListItem: React.FC<IProps> = ({
+  id,
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/components/TodoListItem.tsx]: destructured inside the component

File: src/Modules/DisplayTodo/components/TodoListItem.tsx [hunk 29]
```diff
@@ -0,0 +1,34 @@
...
+    <div className="show-list">
+      <li>
+        <input
+          type="checkbox"
+          className="check-box"
+          checked={completed}
+          onChange={(e) => markTodoCompleted(id, e.target.checked)}
+        />
+        <label htmlFor="task-title" className="task-title">
+          <Link to={`/display/${id}`}>{title}</Link>
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/components/TodoListItem.tsx]: constants ?

File: src/Modules/DisplayTodo/components/index.tsx [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+  showCompleted: boolean;
+  setShowCompleted: (showCompleted: boolean) => void;
+  todos: ToDoData[];
+  markTodoCompleted: (id: number, showCompleted: boolean) => void;
+  loading: boolean;
+  error: string;
+}
+
+const ToDoComponent = ({
+  showCompleted,
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/components/index.tsx]: destructure inside the component!

File: src/Modules/DisplayTodo/components/index.tsx [hunk 31]
```diff
@@ -0,0 +1,52 @@
...
+        showCompleted={showCompleted}
+        setShowCompleted={setShowCompleted}
+      />
+      {loading ? (
+        <div className="loading-spinner">
+          <InfinitySpin color="rgb(246, 162, 67)" width="200" />
+        </div>
+      ) : (
+        <>
+          {error === "" ? (
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/components/index.tsx]: ! error

File: src/Modules/DisplayTodo/constant.ts [hunk 32]
```diff
@@ -0,0 +1,2 @@
+//Todo API
+export const TODO_URL="http://localhost:8000/todo";
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/constant.ts]: create a separate .env file and learn how we can access those variables.
use it here

File: src/Modules/DisplayTodo/index.tsx [hunk 33]
```diff
@@ -0,0 +1,59 @@
...
+
+import { TODO_URL } from "./constant";
+
+const TodoContainer = () => {
+  const [todos, setTodos] = useState<ToDoData[]>([]);
+  const [showCompleted, setShowCompleted] = useState<boolean>(false);
+  const { data, loading, error } = useFetch(TODO_URL);
+
+  useEffect(() => {
+    if (data !== null) setTodos(data);
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/index.tsx]: if(! data)

File: src/Modules/DisplayTodo/index.tsx [hunk 34]
```diff
@@ -0,0 +1,59 @@
...
+  }, [data]);
+
+  const displayTodos = showCompleted
+    ? todos.filter((todo) => todo.completed === true)
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/index.tsx]: todos.filter((todo) => todo.completed)

File: src/Modules/DisplayTodo/index.tsx [hunk 35]
```diff
@@ -0,0 +1,59 @@
...
+    : todos;
+
+  const markTodoCompleted = (id: number, completed: boolean) => {
+    setTodos(
+      todos.map((todo) => {
+        if (todo.id === id) {
+          let tempObj = { ...todo, completed: completed };
+          fetch(`${TODO_URL}/${id}`, {
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTodo/index.tsx:28]: constant?

File: src/data/types/types.ts [hunk 36]
```diff
@@ -0,0 +1,8 @@
+export interface ToDoData{
+    id:number
+    title:string
+    // completed?:boolean
```
Comment (vaibhavchalse99) [inline src/data/types/types.ts]: remove if not using !

File: src/CustomHooks/useFetch.ts (hunk 1 above)
Comment (shubhmore1212) [reply src/CustomHooks/useFetch.ts]: type string assigned
done

File: src/CustomHooks/useFetch.ts (hunk 2 above)
Comment (shubhmore1212) [reply src/CustomHooks/useFetch.ts]: Done

File: src/Modules/AddTodo/components/index.tsx (hunk 3 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/components/index.tsx]: done

File: src/Modules/AddTodo/components/index.tsx (hunk 4 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/components/index.tsx]: Done

File: src/Modules/AddTodo/components/index.tsx (hunk 5 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/components/index.tsx]: done

File: src/Modules/AddTodo/components/index.tsx (hunk 6 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/components/index.tsx]: done

File: src/Modules/AddTodo/components/index.tsx (hunk 7 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/components/index.tsx]: no need of brackets.
State has been changed to pending using constants defined in constants.tsx file

File: src/Modules/AddTodo/components/index.tsx (hunk 8 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/components/index.tsx]: done

Comment (shubhmore1212) [reply src/Modules/AddTodo/components/index.tsx]: done

File: src/Modules/AddTodo/index.tsx (hunk 9 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/index.tsx]: done

File: src/Modules/AddTodo/index.tsx (hunk 10 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/index.tsx]: yes, it should be (e:FormEvent<HtmlFormElement>)

File: src/Modules/AddTodo/index.tsx (hunk 11 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/index.tsx]: yes

File: src/Modules/AddTodo/index.tsx (hunk 12 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/index.tsx]: done

File: src/Modules/AddTodo/index.tsx (hunk 14 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/index.tsx]: Done

File: src/Modules/AddTodo/index.tsx (hunk 13 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/index.tsx]: Done

File: src/Modules/AddTodo/index.tsx (hunk 16 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/index.tsx:67]: Done

File: src/Modules/AddTodo/index.tsx (hunk 15 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/index.tsx]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 17 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 18 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 19 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 20 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 21 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 22 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: Done

File: src/Modules/DisplayTask/index.tsx (hunk 23 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/index.tsx]: Done

File: src/Modules/DisplayTask/index.tsx (hunk 24 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/index.tsx]: Done

File: src/Modules/DisplayTask/index.tsx (hunk 25 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/index.tsx]: Done

File: src/Modules/DisplayTodo/components/AddTodoButton.tsx (hunk 26 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/components/AddTodoButton.tsx]: Done

File: src/Modules/DisplayTodo/components/TodoList.tsx (hunk 27 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/components/TodoList.tsx:8]: Done

File: src/Modules/DisplayTodo/components/TodoListItem.tsx (hunk 28 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/components/TodoListItem.tsx]: done

File: src/Modules/DisplayTodo/components/TodoListItem.tsx (hunk 29 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/components/TodoListItem.tsx]: if constant used it throws err as the url becomes 
**/display/:id/3**

File: src/Modules/DisplayTodo/components/index.tsx (hunk 30 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/components/index.tsx]: done

File: src/Modules/DisplayTodo/components/index.tsx (hunk 31 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/components/index.tsx]: done

File: src/Modules/DisplayTodo/index.tsx (hunk 33 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/index.tsx]: done

File: src/Modules/DisplayTodo/index.tsx (hunk 34 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/index.tsx]: done

File: src/data/types/types.ts (hunk 36 above)
Comment (shubhmore1212) [reply src/data/types/types.ts]: done

File: src/Modules/DisplayTodo/index.tsx (hunk 35 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/index.tsx:28]: done

File: src/Modules/DisplayTodo/constant.ts (hunk 32 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTodo/constant.ts]: done

File: src/CustomHooks/useFetch.ts [hunk 37]
```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from "react";
+
+export const useFetch = (url: any) => {
```
Comment (vaibhavchalse99) [inline src/CustomHooks/useFetch.ts]: use proper type to url

File: src/CustomHooks/useFetch.ts [hunk 38]
```diff
@@ -0,0 +1,28 @@
...
+      .then((res) =>
+        setTimeout(() => {
+          setData(res);
+          setLoading(false);
+        }, 1000)
+      )
+      .catch((err) => {
+        setError(err.message);
+        setLoading(false);
+        console.log(err.message)
```
Comment (vaibhavchalse99) [inline src/CustomHooks/useFetch.ts]: remove logs

File: src/Modules/AddTodo/components/index.tsx [hunk 39]
```diff
@@ -0,0 +1,55 @@
+import React, { ChangeEvent, FormEvent } from "react";
+import { NavLink } from "react-router-dom";
+
+import { ROUTES } from "../../../appContants";
+import { TASK_STATE } from "../../../constant";
+
+interface IProps {
+  title: string;
+  titleHandler: (e: ChangeEvent<HTMLInputElement>) => void;
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: create a separate file named constants.ts and create a type alias for these type and import it here

File: src/Modules/AddTodo/components/index.tsx [hunk 40]
```diff
@@ -0,0 +1,55 @@
...
+  body: string;
+  bodyHandler: (e: ChangeEvent<HTMLTextAreaElement>) => void;
+  handleSubmit: (e: FormEvent<HTMLFormElement>) => void;
+}
+
+const AddTodoComponent: React.FC<IProps> = (props) => {
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: add return type to component

File: src/Modules/AddTodo/components/index.tsx [hunk 41]
```diff
@@ -0,0 +1,55 @@
...
+  const {title,titleHandler,body,bodyHandler,handleSubmit}=props;
+  return (
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx:33]: add blank line

File: src/Modules/AddTodo/components/index.tsx [hunk 42]
```diff
@@ -0,0 +1,55 @@
...
+        <h2>Add Todo</h2>
+        <div>
+          <label htmlFor="title">Task:</label>
+          <input
+            type="text"
+            name="title"
+            value={title}
+            onChange={titleHandler}
+          />
+          <br />
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/components/index.tsx]: remove this br tag, if it is for additional space below input box, add bottom margin instead

File: src/Modules/AddTodo/index.tsx [hunk 43]
```diff
@@ -0,0 +1,91 @@
...
+      <AddTodoComponent
+        title={title}
+        titleHandler={titleHandler}
+        body={body}
+        bodyHandler={bodyHandler}
+        date={date}
+        dateHandler={dateHandler}
+        handleSubmit={handleSubmit}
+      />
+      <Modal
```
Comment (vaibhavchalse99) [inline src/Modules/AddTodo/index.tsx]: move it above <AddTodoComponent/>

File: src/Modules/DisplayTask/components/index.tsx [hunk 44]
```diff
@@ -0,0 +1,102 @@
+import React, { ChangeEvent } from "react";
+import { NavLink } from "react-router-dom";
+
+import Loader from "../../../Shared/components/Loader";
+
+import { ToDoData } from "../../../data/types/types";
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx:6]: blank line after helper files

File: src/Modules/DisplayTask/components/index.tsx [hunk 45]
```diff
@@ -0,0 +1,102 @@
...
+
+const DisplayTaskComponent: React.FC<IProps> = (props) => {
+  const {
+    data,
+    loading,
+    error,
+    markTodoCompleted,
+    modalHandler,
+  } = props;
+  return (
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx:27]: blank line?

File: src/Modules/DisplayTask/components/index.tsx [hunk 46]
```diff
@@ -0,0 +1,102 @@
...
+    <>
+      {loading ? (
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: move this loading condition outside of return 
like 
if(loading){
     return <Loader/>
}

File: src/Modules/DisplayTask/components/index.tsx [hunk 47]
```diff
@@ -0,0 +1,102 @@
...
+                <tbody>
+                  <tr>
+                    <th>ID</th>
+                    <td>:</td>
+                    <td> {data.id}</td>
+                  </tr>
+                  <tr>
+                    <th>Task</th>
+                    <td>:</td>
+                    <td> {data.title}</td>
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: why additional space ?

File: src/Modules/DisplayTask/components/index.tsx [hunk 48]
```diff
@@ -0,0 +1,102 @@
...
+                    <th>Date</th>
+                    <td>:</td>
+                    <td>{data.date}</td>
+                  </tr>
+                  <tr>
+                    <th>Status</th>
+                    <td>:</td>
+                    <td>
+                      <select
+                        defaultValue={data.completed ? "Completed" : "Pending"}
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: I guess we have created constant, can we use it here?

File: src/Modules/DisplayTask/components/index.tsx [hunk 49]
```diff
@@ -0,0 +1,102 @@
...
+                        onChange={markTodoCompleted}
+                      >
+                        <option value={TASK_STATE.COMPLETED}>
+                          {TASK_STATE.COMPLETED}
+                        </option>
+                        <option value={TASK_STATE.PENDING}>
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/components/index.tsx]: better if we destructure it outside the component as it is being used multiple times

File: src/Modules/DisplayTask/index.tsx [hunk 50]
```diff
@@ -0,0 +1,76 @@
...
+    })
+      .then((res) => res.json())
+      .catch((err) => console.log(err));
+    navigate(ROUTES.HOME);
+  };
+
+  const markTodoCompleted = (e: ChangeEvent<HTMLSelectElement>) => {
+    let tempObj = {
+      ...data,
+      completed: e.target.value === TASK_STATE.COMPLETED ? true : false,
```
Comment (vaibhavchalse99) [inline src/Modules/DisplayTask/index.tsx]: e.target.value === TASK_STATE.COMPLETED ? true : false,

can be replace by 

e.target.value === TASK_STATE.COMPLETED

File: src/Shared/components/Loader.tsx [hunk 51]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+import { InfinitySpin } from "react-loader-spinner";
+
+const Loader = () => {
+  return (
+    <div className="loading-spinner">
+      <InfinitySpin color="rgb(246, 162, 67)" width="200" />
```
Comment (vaibhavchalse99) [inline src/Shared/components/Loader.tsx:8]: avoid to use inline css

File: src/Shared/components/Modal.tsx [hunk 52]
```diff
@@ -0,0 +1,34 @@
+interface IProps {
+  isOpen: boolean;
+  title: string;
+  body: string;
+  btn1: string;
+  btn2: string;
+  btn1Handler: any;
```
Comment (vaibhavchalse99) [inline src/Shared/components/Modal.tsx]: add proper type

File: src/Shared/components/Modal.tsx [hunk 53]
```diff
@@ -0,0 +1,34 @@
...
+  btn2Handler: () => void;
+}
+const Modal = (props: IProps) => {
+  const { isOpen, title, body, btn1, btn2, btn1Handler, btn2Handler } = props;
+  return (
+    <>
+      {isOpen ? (
```
Comment (vaibhavchalse99) [inline src/Shared/components/Modal.tsx]: use condition isOpen && ()

File: src/Modules/AddTodo/components/index.tsx (hunk 42 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/components/index.tsx]: Done

File: src/Modules/AddTodo/index.tsx (hunk 43 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/index.tsx]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 44 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx:6]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 45 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx:27]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 46 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 47 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: No need

File: src/Modules/DisplayTask/components/index.tsx (hunk 48 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: Done

File: src/Modules/DisplayTask/components/index.tsx (hunk 49 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/components/index.tsx]: Done

File: src/Shared/components/Modal.tsx (hunk 53 above)
Comment (shubhmore1212) [reply src/Shared/components/Modal.tsx]: Done

File: src/Shared/components/Modal.tsx (hunk 52 above)
Comment (shubhmore1212) [reply src/Shared/components/Modal.tsx]: Done

File: src/Shared/components/Loader.tsx (hunk 51 above)
Comment (shubhmore1212) [reply src/Shared/components/Loader.tsx:8]: can't assign classname or id to library component

File: src/Modules/DisplayTask/index.tsx (hunk 50 above)
Comment (shubhmore1212) [reply src/Modules/DisplayTask/index.tsx]: done

File: src/Modules/AddTodo/components/index.tsx (hunk 41 above)
Comment (shubhmore1212) [reply src/Modules/AddTodo/components/index.tsx:33]: Done

File: src/CustomHooks/useFetch.ts (hunk 37 above)
Comment (shubhmore1212) [reply src/CustomHooks/useFetch.ts]: Done

File: src/CustomHooks/useFetch.ts (hunk 38 above)
Comment (shubhmore1212) [reply src/CustomHooks/useFetch.ts]: Done

**Review by vaibhavchalse99 — APPROVED** [review]

**Pull request merged**